dist/ffau.js

```diff
@@ -593,6 +626,1204 @@ class Ffau {
         return htmlGen.workspaceToCode(this.ffauWorkspace);
     }
 
+    /**
+     * Converts the code string into a Blockly XML string
+     *
+     * @param {string} code - the HTML source code string
+     * @returns {string} - the Blockly XML
+     */
+    codeToXML(code) {
+        let idList = [];
+        let replacerIds = [];
+
+        let cssCommentRegExp = /(?<=^[^"]*?("[^"]*?")*[^"]*?)((?<!\\)\/\*[^*]*?[^\\]\*\/)/g;
+
+        function getReplacerId() {
```

Comment:
Move all these nested functions outside of the class function

Reply:
They're actually serving an important purpose, interacting with variables in the parent function. Removing them/replacing with global variables or making them return objects which are used to redefine the top-scope variables would make it a humongous mess. The accepted answer [here](https://softwareengineering.stackexchange.com/questions/252532/should-i-place-functions-that-are-only-used-in-one-other-function-within-that-f) (note it has more than twice the votes of the highest-voted answer saying 'no') seems to do a pretty good job of giving reasons, plus its upvotes are an indicator that this is accepted practice.